### PR TITLE
Refactor directory location code

### PR DESCRIPTION
**Pull request by DanielSank**
Seems a little cleaner this way.

### CODE REVIEW

File: src/main/scala/org/labrad/registry/Registry.scala [hunk 1]
```diff
@@ -98,6 +98,17 @@ extends ServerActor with Logging {
...
 
+    private def dirForPath(dirs: Seq[String], create: Boolean): store.Dir = {
+      var newDir = curDir
+      for ((dir, i) <- dirs.zipWithIndex) dir match {
+        case ""   => if (i == 0) newDir = store.root
+        case "."  =>
+        case ".." => newDir = store.parent(newDir)
+        case dir  => newDir = _mkDir(newDir, dir, create = create)
+      }
+      curDir
```
Comment (maffoo) [inline src/main/scala/org/labrad/registry/Registry.scala]: should return `newDir` here, not `curDir`

**Comment by DanielSank** [comment]
@maffoo fixed

**Comment by DanielSank** [comment]
@maffoo bump

**Comment by maffoo** [comment]
LGTM

**Pull request merged**